docs/ref/forms/fields.txt

```diff
@@ -275,6 +275,36 @@ fields. We've specified ``auto_id=False`` to simplify the output:
     <div>Sender:<div class="helptext">A valid email address, please.</div><input type="email" name="sender" required></div>
     <div>Cc myself:<input type="checkbox" name="cc_myself"></div>
 
+When a field has help text and :attr:`~django.forms.BoundField.id_for_label`
+returns a value, we associate``help_text`` with the input using
+``aria-describedby``:
+
+    >>> from django import forms
+    >>> class UserForm(forms.Form):
+    ...     username = forms.CharField(max_length=10, help_text='e.g., user@example.com')
+    >>> f = UserForm()
+    >>> print(f.as_div())
+    <div><label for="id_username">Username:</label>
+    <div class="helptext" id="id_username_helptext">e.g., user@example.com</div>
+    <input type="text" name="username" maxlength="10" required aria-describedby="id_username_helptext" id="id_username">
+    </div>
+    
+When adding a custom ``aria-describedby`` attribute, make sure to also include
+the id of the ``help_text`` element (if used) in the desired order.
```

Comment:
I'm not familiar with the `aria-describedby` attribute. The "desired order" is mentioned here and in the comment above, but I, as an aria-ignorant, I have no idea what implications the order may have. For example, what is the difference between:

```
<input type="text" name="username" aria-describedby="custom-description id_username_helptext">
```

and
```
<input type="text" name="username" aria-describedby="id_username_helptext custom-description">
```

If the order of items in the `aria-describedby` is relevant, I think we should explain that in this section. I can relate to Django users trying to do the right thing when it comes to accessibility, but perhaps not being experts in the field, so adding a small explanation here could go a long way in this front.

What do you think?

Reply:
Setting a custom `aria-describedby` should be pretty rare so I thought we’d be safe to assume people interested in this will know how to use the attribute. Particularly so in the even rarer case where order matters because of using both `help_text` and an additional custom description. If we want to cover this specific scenario better, we could do something like:

> […] the id of the `help_text` element (if used) in the desired order. For screen reader users, descriptions will be read in their order of appearance inside `aria-describedby`. For more information about this attribute, refer to [`aria-describedby` documentation on MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby).

Reply:
Thank you @thibaudcolas. I like the adding of:

"For screen reader users, descriptions will be read in their order of appearance inside the `aria-describedby` attribute."

but I would leave out the link to the MDN docs since, in general and as per my understanding, we try to not link or endorse external sites from within the Django docs.

What do you think?